### PR TITLE
Upgrade `develop` to v14.3.1

### LINKED ISSUE #17: Upgrade `develop` to v14.3.1
Pulling in fixes for #2, #9.

### PR DESCRIPTION
**Pull request by joewallwork**
Closes #17.

Linked PRs:
https://github.com/fetch4/geos-chem/pull/5
https://github.com/fetch4/HEMCO/pull/3

Second attempt at #18 using a rebased version of the `dev/giss-gc-14` branch.

### CODE REVIEW

File: config/compiler.gfortran.mk
Comment (joewallwork) [inline config/compiler.gfortran.mk:1]: These changes were already approved as part of #3.

File: config/compiler.intel.mk
Comment (joewallwork) [inline config/compiler.intel.mk:1]: These changes were already approved as part of #3.

File: config/rules.mk
Comment (joewallwork) [inline config/rules.mk:1]: These changes were already approved as part of #3.

File: decks/Makefile
Comment (joewallwork) [inline decks/Makefile:1]: These changes were already approved as part of #3.

File: model/CHEM_DRV.F90
Comment (joewallwork) [inline model/CHEM_DRV.F90:1]: There are several TODOs in this file but I think it'd be fine to follow them up in subsequent issues.

**Comment by joewallwork** [comment]
I think this is pretty much good to go but let's discuss this in the meeting next week. Feel free to leave comments in the meantime.

File: model/CHEM_COM.F90 [hunk 1]
```diff
@@ -48,28 +48,6 @@ module CHEM_COM
...
-      INTEGER :: i_CFCY  = 0
-      INTEGER :: i_CFCZ  = 0
-      INTEGER :: i_SO2   = 0
-      INTEGER :: i_SO4   = 0
-      INTEGER :: i_NIT   = 0
-      INTEGER :: i_OCPI  = 0
-      INTEGER :: i_OCPO  = 0
-      INTEGER :: i_BCPI  = 0
-      INTEGER :: i_BCPO  = 0
-      INTEGER :: i_SOAS  = 0
```
Comment (joewallwork) [inline model/CHEM_COM.F90:71]: This is a change versus the previous implementation rather than GISS modelE itself.

File: model/CLOUDS2_DRV.F90 [hunk 2]
```diff
@@ -215,6 +215,8 @@ subroutine CONDSE
   use CLOUDS, only : mc_up_mf, mc_dd_mf, mc_up_ent, mc_dd_ent, mc_up_det, mc_dd_det
   use CLOUDS, only : mc_dqcond, mc_dqevap, ls_dqcond, ls_dqevap
   use CLOUDS, only : mc_pflx_l, mc_pflx_i, ls_pflx_l, ls_pflx_i
+  use CLOUDS_COM, only : pficu, pflcu, pfilsan, pfllsan
+  use CLOUDS_COM, only : dtrain, dqrcu, dqrlsan, reevapcn, reevapls, cmfmc
```
Comment (joewallwork) [inline model/CLOUDS2_DRV.F90:219]: This is to make these available to GEOS-Chem whilst avoiding a recursive dependency.

File: model/SUBDD.f [hunk 3]
```diff
@@ -1773,6 +1773,13 @@ subroutine parse_subdd
       input_sizes3(k) = lm
       call tijph_defs(diaglists(1,k),nmax_possible,diaglens(k))
 #endif
+#ifdef TRACERS_GC
+      k = k + 1
+      catshapes(k) = 'aijlh'; categories(k) = 'taijlh'
+      input_sizes3(k) = lm
+      call tijlh_defs(diaglists(1,k),nmax_possible,diaglens(k))
+#endif
```
Comment (joewallwork) [inline model/SUBDD.f:1781]: Adds the ability to have GEOS-Chem tracers outputted at hourly timescales.

Backbone for how to translate #23.

File: model/CHEM_DRV.F90
Comment (joewallwork) [inline model/CHEM_DRV.F90:1]: This is the interface itself based on variables defined in `CHEM_COM`.

Comment (joewallwork) [reply model/CHEM_DRV.F90:1]: It mimics `main.F90` to have GEOS-Chem as a CTM.

File: model/CHEM_DRV.F90 [hunk 4]
```diff
@@ -267,9 +271,10 @@ SUBROUTINE DO_CHEM
           State_Met%LAI         (II,JJ) = lai_save(i,j)               
 
           ! Land/water/ice indices [1]
-          State_Met%LWI         (II,JJ) = 1
```
Comment (joewallwork) [inline model/CHEM_DRV.F90:270]: LWI = land, water, ice. (1, 0, 2 resp.)

**Review by AmyOctoCat — APPROVED** [review]
We did a live walkthrough together and this LGTM. :)

**Pull request merged**